src/js/components/AppListComponent.jsx

```diff
@@ -70,13 +70,19 @@ var AppListComponent = React.createClass({
     var sortKey = state.sortKey;
     var props = this.props;
 
-    return lazy(state.apps)
+    var appsSequence = lazy(state.apps);
+
+    if (props.filterText != null && props.filterText != "") {
+      appsSequence = appsSequence
+        .filter(function (app) {
+          return app.id.indexOf(props.filterText) !== -1;
```

Comment:
FTR their examples are REALLY misleading: 

> `Lazy([1, 2, 3]).indexOf(4)  `

I ran a comparison test against a 1000 elements array of progressive 0...1000 values and got 

- `0.142ms` for  `lazy(values).indexOf(500)` 
- `0.033ms` for  `values.indexOf(500)` 

:suspect: